cmd/theatre-envconsul/main.go

```diff
@@ -193,6 +220,30 @@ func mainError(ctx context.Context, command string) (err error) {
 			return errors.Wrap(err, "failed to decode envconsul environment variables")
 		}
 
+		// For every file reference, write the value resolved by envconsul to the path in
+		// filePaths
+		for key, path := range filePaths {
+			if path == pathTBD {
+				// generate file path prefixed by key
+				tempFilePath, err := ioutil.TempFile("", fmt.Sprintf("%s-*", key))
+				if err != nil {
+					return errors.Wrap(err, fmt.Sprintf("failed to write temporary file for key %s", key))
+				}
+				path = tempFilePath.Name()
+			}
+			// ensure the path structure is available
+			os.MkdirAll(filepath.Dir(path), 0644)
+
+			// write file with value of envMap[key]
+			if err := ioutil.WriteFile(path, []byte(envMap[key]), 0644); err != nil {
```

Comment:
For observability and debugging it might be nice to write out a log line whenever we write a file to disk

Reply:
I've copied the schema of the previous log messages.